modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,574 @@
+          twoLineIntersection(bdry_pars[0],
+                              bdry_pars[1],
+                              bdry_pars[2],
+                              p0(1) - p1(1),
+                              p1(0) - p0(0),
+                              -(p0(0) * (p0(1) - p1(1)) + (p1(0) - p0(0)) * p0(1)));
+    }
+  }
+
+  // move nodes
+  for (unsigned int i = 0; i < node_list.size(); i++)
+  {
+    // Only one node in trimmed region
+    // This means the node is on both the trimming boundary and the original external boundary.
+    // In order to keep the shape of the original external boundary, the node is moved along the
+    // original external boundary.
+    if (node_list_flag[i])
+      *(mesh.node_ptr(node_list[i])) = node_list_point[i];
+    // Two nodes in trimmed region, only one is moved
+    // This means the node is in the middle of the trimming boundary.
+    // Just move it along the normal direction of the trimming line.
+    else
+    {
+      const Real x0 = (*(mesh.node_ptr(node_list[i])))(0);
+      const Real y0 = (*(mesh.node_ptr(node_list[i])))(1);
+      (*(mesh.node_ptr(node_list[i])))(0) =
+          (bdry_pars[1] * (bdry_pars[1] * x0 - bdry_pars[0] * y0) - bdry_pars[0] * bdry_pars[2]) /
+          (bdry_pars[0] * bdry_pars[0] + bdry_pars[1] * bdry_pars[1]);
+      (*(mesh.node_ptr(node_list[i])))(1) =
+          (bdry_pars[0] * (-bdry_pars[1] * x0 + bdry_pars[0] * y0) - bdry_pars[1] * bdry_pars[2]) /
+          (bdry_pars[0] * bdry_pars[0] + bdry_pars[1] * bdry_pars[1]);
+    }
+  }
+
+  // Delete the block
+  for (auto elem_it = mesh.active_subdomain_elements_begin(block_id_to_remove);
+       elem_it != mesh.active_subdomain_elements_end(block_id_to_remove);
+       elem_it++)
+    mesh.delete_elem(*elem_it);
+  mesh.contract();
+  mesh.find_neighbors();
+  // Delete zero volume elements
+  for (auto elem_it = mesh.elements_begin(); elem_it != mesh.elements_end(); elem_it++)
+  {
+    if (MooseUtils::absoluteFuzzyEqual((*elem_it)->volume(), 0.0))
+    {
+      for (unsigned int i = 0; i < (*elem_it)->n_sides(); i++)
+      {
+        if ((*elem_it)->neighbor_ptr(i) != nullptr)
+        {
+          boundary_info.add_side((*elem_it)->neighbor_ptr(i),
+                                 ((*elem_it)->neighbor_ptr(i))->which_neighbor_am_i(*elem_it),
+                                 external_boundary_id);
+          boundary_info.add_side((*elem_it)->neighbor_ptr(i),
+                                 ((*elem_it)->neighbor_ptr(i))->which_neighbor_am_i(*elem_it),
+                                 trimming_section_boundary_id);
+        }
+      }
+      mesh.delete_elem(*elem_it);
+    }
+  }
+  mesh.contract();
+  mesh.prepare_for_use();
+
+  // Assign customized outer surface boundary id and name
+  if (external_boundary_id != OUTER_SIDESET_ID)
+    MooseMesh::changeBoundaryId(mesh, OUTER_SIDESET_ID, external_boundary_id, false);
```

Comment:
this should not be part of the lineRemover routine. 
The first time you run this routine, you set the external_bid everywhere, then the second time it is found for the slc_bdry_list